File: octopoints_flutter/lib/service/UserService.dart
import 'package:octopoints_flutter/service/model/User.dart';

abstract class UserService {
  Future<User> createUser(User user);
  Future<int> updateUsers(List<User> users);
  Future<int> deleteUser(User user);
  Future<List<User>> getUsers();
  Future<User> getUserById(int id);
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/service/UserService.kt
package dev.tpavani.octopoints.service

import dev.tpavani.octopoints.service.model.User

interface UserService {
    suspend fun createUser(user:User): User
    suspend fun deleteUser(user: User)
    suspend fun getUsers(): List<User>
    suspend fun updateUser(users: List<User>)
}